src/compiler/moduleNameResolver.ts

```diff
@@ -683,6 +690,16 @@ namespace ts {
         }
     }
 
+    function isEligibleForPathMapping(moduleName: string, paths: MapLike<string[]>): boolean {
+        if (isRootedDiskPath(moduleName) || isUrl(moduleName)) {
+            // check for potential alias otherwise { '*': './src/*' } will match
+            const root = moduleName.slice(0, getRootLength(moduleName));

```

Comment:
You don't need all these checks. The path mapping needs to be done if: moduleName is not relative path or it is absolute path.

Reply:
Did you read the comment?

It would break backwards compatibility and existing configurations.

Reply:
@DanielRosenwasser isn't this change expected breaking change ?

Reply:
Yes, I thought that was the point of this PR in the first place. What exactly are the extra checks meant to protect against?

Reply:
Root (or url) imports failing (resolving to another file) in existing projects failing because they have a `*` mapping.

Reply:
As mentioned this is desired to be a breaking change. Please remove these extra checks